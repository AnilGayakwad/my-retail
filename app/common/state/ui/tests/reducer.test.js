
import { fromJS } from 'immutable';
import UiStateReducer from '../reducer';
import {
  updateUiState,
} from '../actions';

describe('ui Reducer', () => {
  describe('DEFAULT', () => {
    it('returns the initial state', () => {
      const initialState = UiStateReducer(undefined, {}).toJS();
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('UI_STATE', () => {
    it('should update existing "productDetailsPage" object', () => {
      const state = fromJS({
        productDetailsPage: {
          test: 'show be gone',
        },
      });
      const action = updateUiState({ test: 'passed' });
      expect(UiStateReducer(state, action)).toMatchSnapshot();
    });
  });
});
