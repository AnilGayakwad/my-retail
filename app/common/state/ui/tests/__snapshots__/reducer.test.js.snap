// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui Reducer DEFAULT returns the initial state 1`] = `
Object {
  "productDetailsPage": Object {
    "imageToDisplay": 0,
    "photoIndex": 0,
    "quantitySelected": 1,
    "showLargeImages": false,
  },
}
`;

exports[`ui Reducer UI_STATE should update existing "productDetailsPage" object 1`] = `
Immutable.Map {
  productDetailsPage: Immutable.Map {
    test: "passed",
  },
}
`;
